src/compiler/scala/tools/nsc/tasty/TreeUnpickler.scala

```diff
@@ -498,6 +498,9 @@ class TreeUnpickler[Tasty <: TastyUniverse](
           }
         }
       }.ensuring(isSymbol(_), s"${ctx.classRoot}: Could not create symbol at $start")
+      if (tag == VALDEF && flags.is(Enum)) {
+        sym.updateAttachment(new symbolTable.DottyEnumSingleton(sym.name.toString))
```

Comment:
I wonder if we could (intelligently) reuse the `JAVA_ENUM` flag for this. A actual *Java* enum would have `JAVA_ENUM` *and* `JAVA`, whereas a Dotty enum would only have `JAVA_ENUM`.

Perhaps you could audit uses of `JAVA_ENUM` / `isJavaEnum` to see whether that would make sense (simplify some code paths or not affect them) or not (making some code paths more complex).

That could definitely done in a separate PR, though.

Reply:
if there was a way to get the original name of the symbol before it is mangled to a class name I'd be happy

Reply:
`originalName`?

Reply:
no that is still an inner class name, e.g. `Colour$Red`